cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -457,8 +554,10 @@ func (dc *DownstreamController) deviceUpdated(device *v1alpha1.Device) {
 				}
 				dc.deviceDeleted(deletedDevice)
 
+			} else if isProtocolConfigUpdated(&cachedDevice.Spec.Protocol, &device.Spec.Protocol) {
+				dc.updateProtocolInConfigMap(device)
```

Comment:
A quick question, when an error occurs in updateProtocolInConfigmap, will the following procedure continue?

Reply:
The function only has if else cases, so when this else if ends, the function exits.